core/services/keeper/upkeep_executer.go

```diff
@@ -239,51 +239,69 @@ func (executer *UpkeepExecuter) execute(upkeep UpkeepRegistration, headNumber in
 	}
 
 	// TODO: Remove in
 	// https://app.clubhouse.io/chainlinklabs/story/6065/hook-keeper-up-to-use-tasks-in-the-pipeline
 	elapsed := time.Since(start)
-	pipeline.PromPipelineTaskExecutionTime.WithLabelValues(fmt.Sprintf("%d", executer.job.ID), executer.job.Name.String, "", job.Keeper.String()).Set(float64(elapsed))
+	pipeline.PromPipelineTaskExecutionTime.WithLabelValues(fmt.Sprintf("%d", ex.job.ID), ex.job.Name.String, "", job.Keeper.String()).Set(float64(elapsed))
 	var status string
 	if runErrors.HasError() || err != nil {
 		status = "error"
-		pipeline.PromPipelineRunErrors.WithLabelValues(fmt.Sprintf("%d", executer.job.ID), executer.job.Name.String).Inc()
+		pipeline.PromPipelineRunErrors.WithLabelValues(fmt.Sprintf("%d", ex.job.ID), ex.job.Name.String).Inc()
 	} else {
 		status = "completed"
 	}
-	pipeline.PromPipelineRunTotalTimeToCompletion.WithLabelValues(fmt.Sprintf("%d", executer.job.ID), executer.job.Name.String).Set(float64(elapsed))
-	pipeline.PromPipelineTasksTotalFinished.WithLabelValues(fmt.Sprintf("%d", executer.job.ID), executer.job.Name.String, "", job.Keeper.String(), status).Inc()
+	pipeline.PromPipelineRunTotalTimeToCompletion.WithLabelValues(fmt.Sprintf("%d", ex.job.ID), ex.job.Name.String).Set(float64(elapsed))
+	pipeline.PromPipelineTasksTotalFinished.WithLabelValues(fmt.Sprintf("%d", ex.job.ID), ex.job.Name.String, "", job.Keeper.String(), status).Inc()
 }
 
-func (executer *UpkeepExecuter) constructCheckUpkeepCallMsg(upkeep UpkeepRegistration) (ethereum.CallMsg, error) {
+func (ex *UpkeepExecuter) constructCheckUpkeepCallMsg(upkeep UpkeepRegistration) (ethereum.CallMsg, error) {
 	checkPayload, err := RegistryABI.Pack(
 		checkUpkeep,
 		big.NewInt(int64(upkeep.UpkeepID)),
 		upkeep.Registry.FromAddress.Address(),
 	)
 	if err != nil {
-		return ethereum.CallMsg{}, err
+		return ethereum.CallMsg{}, errors.Wrap(err, "failed to pack payload of check upkeep")
 	}
 
 	to := upkeep.Registry.ContractAddress.Address()
-	gasLimit := executer.config.KeeperRegistryCheckGasOverhead() + uint64(upkeep.Registry.CheckGas) +
-		executer.config.KeeperRegistryPerformGasOverhead() + upkeep.ExecuteGas
-	msg := ethereum.CallMsg{
+	gasLimit := ex.config.KeeperRegistryCheckGasOverhead() + uint64(upkeep.Registry.CheckGas) +
+		ex.config.KeeperRegistryPerformGasOverhead() + upkeep.ExecuteGas
+
+	return ethereum.CallMsg{
 		From: utils.ZeroAddress,
 		To:   &to,
 		Gas:  gasLimit,
 		Data: checkPayload,
+	}, nil
+}
+
+// logRevertReason logs the given error with a log level depends on this given error context
+// Default log level is error. Mapping between a reason and its log level:
+//  - "upkeep not needed": Debug
+func (ex *UpkeepExecuter) logRevertReason(logger *logger.Logger, err error) {
+	revertReason, err2 := eth.ExtractRevertReasonFromRPCError(err)
+	if err2 != nil {
+		logger.WithError(err).Errorf("UpkeepExecuter: checkUpkeep call failed with unknown reason, err2: %v", err2)
```

Comment:
can we change this to something like "failed to extract revert reason" or something like that?

Reply:
Done